src/app/api/wallet/service.ts

```diff
@@ -0,0 +1,63 @@
+import prisma from "db";
+
+export type UserWallet = {
+  id: string;
+  user_id: string;
+  address: string;
+  created_at: Date;
+};
+
+export async function saveUserWallet(
+  userId: string,
+  address: string,
+): Promise<UserWallet | null> {
+  try {
+    // Check if user wallet exists for the given user ID
+    const existingWallet = await prisma.userWallet.findUnique({
+      where: {
+        user_id: userId,
+      },
+    });
+
+    if (existingWallet) {
+      // User wallet exists, update it
+      const updatedWallet = await prisma.userWallet.update({
+        where: {
+          user_id: userId,
+        },
+        data: {
+          address: address,
+        },
+      });
+      return updatedWallet;
+    } else {
+      // User wallet doesn't exist, create it
+      const newWallet = await prisma.userWallet.create({
+        data: {
+          user_id: userId,
+          address: address,
+        },
+      });
+      return newWallet;
+    }
```

Comment:
can you double check if prisma doesn't have a createOrUpdate? I have a feeling that it does, prob we could cut this logic if they offer that functionality